docs/guide/topics/error_handling.rst

```diff
@@ -0,0 +1,361 @@
+.. currentmodule:: discord
+
+
+==========================
+Errors and Error Handling
+==========================
+
+Handling Errors
+===============
+
+When an error ocurrs in a command or an event, it will, by default, show an error in the console to denote that the error 
+ocurred and has been silently ignored. In order to avoid this, an error handler is needed.
+
+In :ref:`Commands <discord_ext_commands>`
+-----------------------------------------
+
+All errors that are raised by a command derive from :exc:`~.ext.commands.CommandError`. If an error that occurs during a command does
+not inherit from :exc:`~.ext.commands.CommandError`, it will automatically get wrapped in a :exc:`~.ext.commands.CommandInvokeError`, 
+to retrieve/unwrap this error we need to access it's ``.original`` attribute.
+
+**Example**
+
+.. code-block:: python3
+
+    if isinstance(error, commands.CommandInvokeError):
+         error = error.original
+
+Global Error handlers
+~~~~~~~~~~~~~~~~~~~~~
+
+You can create global error handlers by creating :func:`.on_command_error` events, which work like any other event in 
+the :ref:`discord-api-events`, these are called for every exception that occurs in a command, before invoke hooks, command
+converters, etc.
+
+**Example**
+
+.. code-block:: python3
+    :emphasize-lines: 1, 2
+
+    @bot.listen()
+    async def on_command_error(ctx, error):
```

Comment:
```suggestion
    async def on_command_error(ctx: commands.Context, error: commands.CommmandError) -> None:
```
I believe since we are now a strongly typed library we should be suggesting typehints here.